server/multitenancy/tenant_resolver.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { KibanaRequest } from '../../../../src/core/server';
+import { SecurityPluginConfigType } from '..';
+import { SecuritySessionCookie } from '../session/security_cookie';
+import { isEmpty, findKey, cloneDeep } from 'lodash';
+
+/**
+ * Resovles the tenant the user is using.
```

Comment:
is this jsdoc for resolveTenat() method below, if so should we move to above the method?

Reply:
ah, yes, good catch